src/lib.rs

```diff
@@ -241,14 +259,17 @@ pub trait ForumUserRegistry<AccountId> {
 }
 
 /// Convenient composite time stamp 
-struct BlockchainTimestamp<BlockNumber, Moment> {
+//#[cfg_attr(feature = "std", derive(Serialize, Deserialize, Debug))]
+#[derive(Encode, Decode, Default, Clone, PartialEq, Eq)]
+pub struct BlockchainTimestamp<BlockNumber, Moment> {
```

Comment:
We have something a bit like that in the storage module; it'd be great to extract this into something more re-usable.

There are a few moments where we want to record the block number/time.

Reply:
100% agreed.

Reply:
Yes, I agree, but this is a standalone repo and crate, so I cannot reuse before we actually create self-contained package with all of the common substrate things we want to share across all of our modules. This may be of value to others also.